src/modules/Modal/Modal.js

```diff
@@ -1,48 +1,67 @@
-import React, { Component, PropTypes } from 'react'
-import classNames from 'classnames'
+import React, { PropTypes } from 'react'
+import cx from 'classnames'
 import META from '../../utils/Meta'
-import ModalContent from './ModalContent'
 import ModalHeader from './ModalHeader'
-import ModalFooter from './ModalFooter'
-
-export default class Modal extends Component {
-  static propTypes = {
-    children: PropTypes.any,
-    className: PropTypes.string,
-    settings: PropTypes.object,
-  }
-
-  state = { isShown: false }
-
-  static Content = ModalContent
-  static Header = ModalHeader
-  static Footer = ModalFooter
-
-  showModal = () => {
-    this.setState({ isShown: true })
-  }
-
-  hideModal = () => {
-    this.setState({ isShown: false })
-  }
-
-  static _meta = {
-    name: 'Modal',
-    type: META.type.module,
-  }
-
-  render() {
-    const classes = classNames(
-      'ui',
-      this.props.className,
-      'modal',
-      { 'transition visible active': this.state.isShown },
-    )
-
-    return (
-      <div {...this.props} className={classes}>
-        {this.props.children}
-      </div>
-    )
-  }
+import ModalContent from './ModalContent'
+import ModalActions from './ModalActions'
+import {
+  getUnhandledProps,
+  useKeyOnly,
+} from '../../utils/propUtils'
+
+function Modal(props) {
+  const {
+    children, className,
+    active, basic, size, long,
+  } = props
+
+  const classes = cx('ui modal',
```

Comment:
For consistency, let's put the `ui` class first and the `modal` class just before the user's `className`.